packages/front-end/pages/settings/index.tsx

```diff
@@ -193,8 +191,15 @@ const GeneralSettingsPage = (): React.ReactElement => {
       const newVal = { ...form.getValues() };
       Object.keys(newVal).forEach((k) => {
         const hasExistingMetrics = typeof settings?.[k] !== "undefined";
-        newVal[k] = settings?.[k] || newVal[k];
-
+        if (k === "metricDefaults") {
+          // Metric defaults are nested, so take existing metric defaults only if
+          // they exist and are not empty
+          Object.keys(newVal[k]).forEach((kk) => {
+            newVal[k][kk] = settings?.[k]?.[k] ?? newVal[k][kk];
```

Comment:
This is accomplishing the same as if you used the spread operator (`...`), with `settings` spread first. Any reason you wouldn't want to do it that way?

Reply:
It needs to spread settings?.[k] which could work.